EFCache.Redis/RedisCache.cs

```diff
@@ -1,38 +1,152 @@
+using RedLockNet.SERedis;
+using RedLockNet.SERedis.Configuration;
+using StackExchange.Redis;
 using System;
 using System.Collections.Generic;
-using System.Data.Common;
+using System.Diagnostics;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using System.Security.Cryptography;
 using System.Text;
 using System.Text.RegularExpressions;
-using StackExchange.Redis;
+using System.Threading;
+using RedLockNet;
 
 namespace EFCache.Redis
 {
-	public class RedisCache : IRedisCache
+	/// <inheritdoc cref="IRedisCache" />
+	/// <inheritdoc cref="IDisposable"/>
+	/// <summary>
+	/// This class implement ICache via IRedisCache. Provides for (potentially) atomic management of sql query result caching
+	/// via a mechanism of
+	/// a) Redis sets (one per entity set) containing references to keys for queries referencing
+	/// that entity set
+	/// b) Strings keyed on the hash of the sql query storing the result set
+	/// This class makes use of the RedLock algorithm https://redis.io/topics/distlock to offer a
+	/// mechanism for two phase-style commit via locks and lua scripts for writes to ensure
+	/// changes are atomic and can be serialized per entity set.
+	/// </summary>
+	public class RedisCache : IRedisCache, IDisposable
 	{
+		#region Private Vars
+		// Prefixes for cache keys
 		private const string DefaultCacheIdentifier = "__EFCache.Redis_EntitySetKey_";
+		private readonly string _cacheIdentifier;
```

Comment:
@jkdufair is `_cacheIdentifier` or `_statsIdentifier ` ever actually set? I see they are used in the util methods, but not where they are set.

Reply:
In the constructor

Reply:
@jkdufair ah i see now